### PR TITLE
sassc: modernize

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  *sassc/***

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

File: recipes/sassc/all/conanfile.py [hunk 1]
```diff
@@ -21,31 +24,31 @@ class SasscConan(ConanFile):
...
-
     def config_options(self):
         del self.settings.compiler.libcxx
         del self.settings.compiler.cppstd
 
-    def configure(self):
-        if not self._is_msvc and self.settings.os not in ["Linux", "FreeBSD", "Macos"]:
+    def package_id(self):
+        del self.info.settings.compiler
+        del self.info.settings.build_type
```
Comment (uilianries) [inline recipes/sassc/all/conanfile.py:33]:
```suggestion
```

Please, keep build_type: https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md#settings

File: recipes/sassc/all/conanfile.py [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import get, replace_in_file, chdir, save
+from conan.tools.microsoft import is_msvc
+from conans import AutoToolsBuildEnvironment, tools, MSBuild
 import os
 
-required_conan_version = ">=1.33.0"
+required_conan_version = ">=1.52.0"
```
Comment (uilianries) [inline recipes/sassc/all/conanfile.py:8]:
```suggestion
required_conan_version = ">=1.47.0"
```
1.47.0 should be enough here.

**Review by jwillikers — APPROVED** [review]

File: recipes/sassc/all/conanfile.py [hunk 3]
```diff
@@ -67,25 +69,34 @@ def _build_msbuild(self):
...
+        self.cpp_info.frameworkdirs = []
+        self.cpp_info.libdirs = []
+        self.cpp_info.resdirs = []
+        self.cpp_info.includedirs = []
+
+        bin_folder = os.path.join(self.package_folder, "bin")
+        # In case need to find packaged tools when building a package
+        self.buildenv_info.append("PATH", bin_folder)
+        # In case need to find packaged tools at runtime
+        self.runenv_info.append("PATH", bin_folder)
```
Comment (uilianries) [inline recipes/sassc/all/conanfile.py:100]:
```suggestion
```
In the past I thought it was needed, but after reading docs and checking those generated scripts, PATH is automatically filled by Conan: 

https://docs.conan.io/en/latest/reference/conanfile/tools/env/virtualbuildenv.html
https://docs.conan.io/en/latest/reference/conanfile/tools/env/virtualrunenv.html

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**